src/contact.rs

```diff
@@ -1317,6 +1317,22 @@ impl Contact {
         }
     }
 
+    /// Returns if the contact profile title should display a green checkmark.
+    ///
+    /// This generally should be consistent with the 1:1 chat with the contact
+    /// so 1:1 chat with the contact and the contact profile
+    /// either both display the green checkmark or both don't display a green checkmark.
+    pub async fn is_profile_verified(&self, context: &Context) -> Result<bool> {
+        let contact_id = self.id;
+
+        if let Some(chat_id) = ChatId::lookup_by_contact(context, contact_id).await? {
+            Ok(chat_id.is_protected(context).await? == ProtectionStatus::Protected)
```

Comment:
ftr, this branch may never be used by UI: if there is a chat, the UI may render things differently and check chat_is_protected() directly - as eg. in practise, one-to-one profile and group-profile are similar (and for group-profile, contact_is_verified() does not help

not a blocker and i do not suggest changes, just saying :)

Reply:
stop looking at 1:1chat from contact profile does not make sense from UX. 1:1 chat profiles partly show chat-functions (disappearing messages, mute, media), party contact functions (block, edit name) - the concrete set depends on UI

still, this pr moves some of the logic to core so that we can easier adapt things as needed